apps/emqx/test/emqx_common_test_helpers.erl

```diff
@@ -875,11 +890,9 @@ setup_node(Node, Opts) when is_map(Opts) ->
                         integer_to_list(erlang:unique_integer())
                     ]),
                     Cookie = atom_to_list(erlang:get_cookie()),
-                    os:putenv("EMQX_NODE__DATA_DIR", NodeDataDir),
-                    os:putenv("EMQX_NODE__COOKIE", Cookie),
+                    set_env_once("EMQX_NODE__DATA_DIR", NodeDataDir),
+                    set_env_once("EMQX_NODE__COOKIE", Cookie),
```

Comment:
Maybe we should do that only for `emqx` app, not for **every** app? I.e. change function clause to `fun(emqx) ->`.

Reply:
Hmm these setups are so finicky, I wonder if doing it only for `emqx` won't break some other suite that relies on this running for other apps like `emqx_conf` :thinking: 

I'll make this change and see if CI breaks.

Reply:
Hmm apparently that's exactly what happened with Pulsar Producer's suite.  Both locally and in CI I'm consistently getting failures (timeout waiting for the bridge post update to finish) after this change.  I'll have to revert it. :see_no_evil: :speak_no_evil: